src/replicator.js

```diff
@@ -93,15 +95,27 @@ class Replicator {
     this._pubsub = new Pubsub(this.ipfs, (await this.ipfs.id()).id)
     // Passes default cache but with fixed path instead of path based on
     // orbitdb/ipfs id which can change on page load
-    const cachePath = path.join(opts.orbitPath || './orbitdb', '/cache')
-    const levelDown = OdbStorage(null, {})
-    const cacheStorage = await levelDown.createStore(cachePath)
-    const cache = new OdbCache(cacheStorage)
 
+    const levelDown = OdbStorage(null, {})
     const keystorePath = path.join(opts.orbitPath || './orbitdb', '/keystore')
     const keyStorage = await levelDown.createStore(keystorePath)
     const keystore = new OdbKeystore(keyStorage)
 
+    let cache
+    const cachePath = path.join(opts.orbitPath || './orbitdb', '/cache')
+
+    if (opts.iframeCache) {
+      const iframe = document.querySelector('iframe')
+      const postMessage = iframe.contentWindow.postMessage.bind(iframe.contentWindow)
+
+      const cacheStorageProxy = await SharedCache.createOrbitStorageProxy(cachePath, { postMessage })
+
+      cache = new OdbCache(cacheStorageProxy)
+    } else {
+      const cacheStorage = await levelDown.createStore(cachePath)
+      cache = new OdbCache(cacheStorage)
+    }
```

Comment:
Doesn't seem right to have the document query selector in this class. Should be abstracted away to happen only in 3box.js maybe? The `cacheStorageProxy` could be sent as an option to the Replicator class perhaps.

Reply:
:thinking: I will look into it. Passing the cacheStorageProxy as a dependency is good idea as well.